internal/network/dhcp_param_file.go

```diff
@@ -0,0 +1,47 @@
+package network
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/go-openapi/swag"
+	"github.com/openshift/assisted-service/internal/common"
+	"gopkg.in/yaml.v3"
+)
+
+type vip struct {
+	Name       string `yaml:"name"`
+	MacAddress string `yaml:"mac-address"`
+	IpAddress  string `yaml:"ip-address"`
+}
+type vips struct {
+	APIVip     *vip `yaml:"api-vip"`
+	IngressVip *vip `yaml:"ingress-vip"`
+}
+
+func generateOpenshiftDhcpParamFileContents(cluster *common.Cluster) ([]byte, error) {
+	if swag.BoolValue(cluster.VipDhcpAllocation) && cluster.APIVip != "" && cluster.IngressVip != "" {
```

Comment:
Shouldn't we return an error in case 
wag.BoolValue(cluster.VipDhcpAllocation) is true but one of (or both) other condition is false?

Reply:
OK.  Will change